components/nexusoperations/events_test.go

```diff
@@ -140,3 +142,113 @@ func TestCherryPick(t *testing.T) {
 		}
 	})
 }
+
+func TestTerminalStatesDeletion(t *testing.T) {
+	setup := func(t *testing.T) (*hsm.Node, nexusoperations.Operation, int64) {
+		node := newOperationNode(t, &hsmtest.NodeBackend{}, mustNewScheduledEvent(time.Now(), time.Hour))
+		op, err := hsm.MachineData[nexusoperations.Operation](node)
+		require.NoError(t, err)
+		eventID, err := hsm.EventIDFromToken(op.ScheduledEventToken)
+		require.NoError(t, err)
+		return node, op, eventID
+	}
+
+	applyEventAndCheckDeletion := func(
```

Comment:
You can just inline this in the loop body, there's no value in extracting to a function.

Reply:
Done!

Reply:
Make sense. I'm not strongly opinionated here. Changed it